### ISSUE TITLE
Bug: serializing/deserializing structs stopped working after upgrading from anchor 0.18 to 0.22

### ISSUE THREAD

**Issue by WrRaThY** [project-serum/anchor#1569]
tl;dr - the bug is reproduced here: https://github.com/WrRaThY/anchor-bug-report
there are 2 blocks of code with `//COMMENT:` that lets you easily switch between versions to reproduce it

description:
Hey
looks like with the new anchor version there is something wrong with serializing number fields if they have different ranges.
My example is based on
```
#[account(zero_copy)]
#[derive(Default)]
pub struct Config {
    pub timestamp1: i64,
    pub some_u32: u32,
    pub timestamp2: i64,
}
```
with the old version `timestamp1` is the same as `timestamp2` (`1646683332`), with anchor 0.22 - `timestamp2` has some crazy value (`7072451057808310272`).

after some testing - I have some additional input that may help fix this bug:

- looks like anchor JS library version does not impact the result, only anchor rust.
- if you move `some_u32` as the last variable - it works again (but I don't have that luxury, but I have thousands of accounts on mainnet with that structure))
- it only fails for `AccountLoader`, the normal `Account` doesn't have that issue

**Comment by WrRaThY** [comment]
looks like I'm dumb and I didn't understand the breaking change from anchor 0.20.0
adding `[repr(packed)]` to the account fixes the problem
